applications/plugins/ArticulatedSystemPlugin/src/ArticulatedSystemPlugin/ArticulatedSystemMapping.inl

```diff
@@ -76,6 +75,17 @@ void ArticulatedSystemMapping<TIn, TInRoot, TOut>::init()
     {
         m_fromRootModel = this->getFromModels2()[0];
         msg_info() << "Root Model found : Name = " << m_fromRootModel->getName();
+
+        sofa::Size rootSize = m_fromRootModel->getSize();
+        if(d_indexFromRoot.isSet()) {
+            if(d_indexFromRoot.getValue() >= rootSize)
+            {
+                msg_warning() << d_indexFromRoot.getName() << ", " << d_indexFromRoot.getValue() << ", is larger than input2's size, " << rootSize << ". Using default value instead.";
+                d_indexFromRoot.setValue(rootSize - 1);
+            }
+        } else {
+            d_indexFromRoot.setValue(rootSize - 1); // default is last index
+        }
```

Comment:
```suggestion
        const sofa::Size rootSize = m_fromRootModel->getSize();
        if(d_indexFromRoot.isSet()) 
        {
            if(d_indexFromRoot.getValue() >= rootSize)
            {
                msg_warning() << d_indexFromRoot.getName() << ", " << d_indexFromRoot.getValue() << ", is larger than input2's size, " << rootSize << ". Using default value instead.";
                d_indexFromRoot.setValue(rootSize - 1);
            }
        } 
        else 
        {
            d_indexFromRoot.setValue(rootSize - 1); // default is last index
        }
```
(reformatting braces)
and maybe saying about what is this "default value" is, aka  `Using the default value instead which in this case will be "<< rootSize-1;` ?